pkg/version/update.go

```diff
@@ -150,15 +156,12 @@ func RunUpdateChecker(
 
 		if err == nil {
 			responseCallback(ctx, updateResponse)
-			// TODO(forrest): similar to the below TODO, shove this in a cache and persit on shutdown.
-			err = writeNewLastCheckTime(cfg.Update.CheckStatePath)
+			err = store.WriteLastUpdateCheck(time.Now())
 			log.Ctx(ctx).WithLevel(logger.ErrOrDebug(err)).Err(err).Msg("Completed update check")
 		}
 	}
 
-	// TODO(forrest) [efficiency]: rather than repeatedly reading a file, set the value in a cache and flush the cache
-	// to disk when this service shutwdown, no reason for this IO.
-	lastCheck, err := readLastCheckTime(cfg.Update.CheckStatePath)
+	lastCheck, err := store.ReadLastUpdateCheck()
```

Comment:
what changed your mind about the TODO statement?
> // TODO(forrest) [efficiency]: rather than repeatedly reading a file, set the value in a cache and flush the cache
	// to disk when this service shutwdown, no reason for this IO.